src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs

```diff
@@ -95,6 +98,14 @@ protected override void ExecuteCore()
             {
                 frameworkReferenceMap.TryGetValue(knownFrameworkReference.Name, out ITaskItem frameworkReference);
 
+                if (frameworkReference != null &&
+                    frameworkReference.ItemSpec.StartsWith(WindowsDesktopFramework) &&
+                    !RuntimeEnvironment.GetRuntimeIdentifier().StartsWith("win"))
+                {
+                    Log.LogError(Strings.WindowsDesktopFrameworkRequiresWindows);
+                    continue;
+                }
```

Comment:
I read this wrong the first time and now I'm not sure how it works. Studying.

Reply:
(Aside:  really think this Target needs to be renamed. This is like the 4th time I think this is what turns framework references into individual references, but that's elsewhere.)

Reply:
`MakeFrameworkReferencesWorkPlease`

Reply:
So we're looping through all the known framework references here, and I don't get how this error wouldn't trigger always on non-Windows since I would expect windowsdekwtop to be always in this list. @dsplaisted, @peterhuene  What am I missing?

On the first read, I thought this was processing the resolved framework references, not the known framework references.

Reply:
The `frameworkReference != null` check looks like it stops the error from happening on non-Windows.

However, I think it will still download the targeting and runtime packs, which we want to avoid.

Reply:
The logic is a little confusing.  `frameworkReferenceMap` is mapping the `FrameworkReference` names to their item specs.  We're iterating the *known* frameworks and checking each to see if it is being referenced.

Thus `frameworkReference` is only non-null if it's being referenced by the project.

Actually, now that I think about it, this should probably have a check for skipping `WindowsDesktop` on non-Windows when *not referenced*.  We could also just exclude it as a known framework reference in core-sdk based on the SDK RID, perhaps.

Reply:
> frameworkReferenceMap is mapping the FrameworkReference names to their item specs

This is where I was confused, thanks.

So the map only has the direct FrameworkReferences, right? Because this runs before we can find transitive ones in the assets file? We need to error on the transitive case as well.

Reply:
> However, I think it will still download the targeting and runtime packs, which we want to avoid.

Yes, we need to avoid this. I see that below, the adding of known framework reference to download is not conditioned on the frameworkReference != null, which it can't be because we have to eagerly download all in case we get transitive references, right?

Reply:
It will be tricky to test the transitive FrameworkReference failure, because we won't be able to build a package or a project that has a WindowsDesktop FrameworkReference on the same machine.

As is, it should [fail](https://github.com/dotnet/sdk/blob/d6749cfae26412c188e26c247b15919afccef127/src/Tasks/Microsoft.NET.Build.Tasks/ResolveTargetingPackAssets.cs#L149) with NETSDK1073: The FrameworkReference 'Microsoft.WindowsDesktop.App' was not recognized.

Reply:
> As is, it should fail with NETSDK1073: The FrameworkReference 'Microsoft.WindowsDesktop.App' was not recognized.

I think it would be useful. and much clearer if the error was NETSDK1100 in transitive case as well. Bonus points if we can have an error that "blames" the package/project that pulled in WindowsDesktop, though that would be involved and something I'd definitely handle separately. If it's going to fail with 1073 as is, I can maybe live with that for now too with an issue tracking a better diagnostic. depending on how involved fixing it is.

> It will be tricky to test the transitive FrameworkReference failure, because we won't be able to build a package or a project that has a WindowsDesktop FrameworkReference on the same machine.

Ugh, good point. We should at least manually test.

Reply:
Thanks for the feedback, both.  I will investigate this.

Reply:
Perhaps we should mark the `FrameworkReference` as windows only (copying the metadata from the known framework reference) and then again check the platform when resolving targeting pack assets to handle the transitive case? Thoughts?

Reply:
Scratch that, we don't output those currently.   Still looking for a minimally invasive way to handle the transitive case.

Reply:
I've been staring at the targets and tasks for a while now and I think I'll need more time than we have for P6 to gracefully handle the transitive case.

I'm going to push up the feedback changes and proceed and I'll file an issue to improve the transitive experience.